cpp/oneapi/dal/backend/primitives/lapack/test/syevd_dpc.cpp

```diff
@@ -128,6 +128,46 @@ class syevd_test : public te::float_algo_fixture<Float> {
         }
     }
 
+    void check_eigvals_with_eigen(const la::matrix<Float>& s,
+                                  const la::matrix<Float>& eigvecs,
+                                  const la::matrix<Float>& eigvals) const {
+        INFO("convert results to float64");
+        const auto s_f64 = la::astype<double>(s);
+        const auto eigvals_f64 = la::astype<double>(eigvals);
+        const auto eigvecs_f64 = la::astype<double>(eigvecs);
+        std::int64_t row_count = s.get_row_count();
+        std::int64_t column_count = s.get_column_count();
+        const Float* data = s.get_data();
+
+        Eigen::Matrix<Float, Eigen::Dynamic, Eigen::Dynamic> eigen_matrix(row_count, column_count);
+        for (int i = 0; i < eigen_matrix.rows(); ++i) {
```

Comment:
An Eigen matrix can also be created from a non-owned pointer for float/double data. I think something along the lines of `Eigen::map< Eigen::Matrix<info, incl. row/col major> , alignment, stride >`.